flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/junit/extensions/retry/strategy/RetryOnExceptionStrategy.java

```diff
@@ -37,6 +40,12 @@ public RetryOnExceptionStrategy(
     @Override
     public void handleException(String testName, int attemptIndex, Throwable throwable)
```

Comment:
Could we improve the JavaDoc for this method in `RetryStrategy`? Essentially, `TestAbortedException` is used for aborting a current test retry run. Any other exception will result in a failure of the current test run. calling `stopFollowingAttempts()` will result in any succeeding retries to not be executed (see [RetryTestExecutionExtension:53](https://github.com/apache/flink/blob/78b231f60aed59061f0f609e0cfd659d78e6fdd5/flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/junit/extensions/retry/RetryTestExecutionExtension.java#L53)).

Reply:
Sounds good -- I'll summarize the JUnit behaviour (actually documented in https://junit.org/junit5/docs/5.0.2/api/org/junit/jupiter/api/extension/TestExecutionExceptionHandler.html#handleTestExecutionException-org.junit.jupiter.api.extension.ExtensionContext-java.lang.Throwable-)

Reply:
Here's a top: Updating the JavaDoc could be something that can be moved into its own commit because it's kind of unrelated (i.e. if we want to revert the JUnit migration commit, this change is still valid and shouldn't be reverted).